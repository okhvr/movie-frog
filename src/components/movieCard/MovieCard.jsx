import React, { Component } from 'react';
import { Link } from "react-router-dom";

import './style.scss';

export default class MovieCard extends Component {

    render() {
        const {movie} = this.props;
        return (
            <Link to={'/movie/' + movie.id}>
                <div className="card margins"style={{width: '18rem'}}>
                    <img src={movie.poster_path} className="card-img-top"/>
                    <div className="card-body">
                    <h5 className="card-title">{movie.title}<span className="badge badge-secondary">{movie.release_date}</span></h5>
                    <h6 className="card-subtitle mb-2 text-muted">Card subtitle</h6>
                    </div>
                </div>
            </Link>
        );
    }
}
