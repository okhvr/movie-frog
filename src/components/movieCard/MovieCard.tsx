import { Component } from 'react';
import * as React from 'react';
import { Link } from 'react-router-dom';
import styled from 'styled-components';
import { MovieType } from '../../types';

type MyProps = {
    movie: MovieType,
};
const StyledContainer = styled.div.attrs({
    className: "card",
  })`
    width: 18rem;
    margin: 20px;
  `;

export default class MovieCard extends Component {
    props: MyProps;

    render() {
        const {movie} = this.props;
        return (
            <Link to={'/movie/' + movie.id}>
                <StyledContainer>
                    <img src={movie.poster_path} className="card-img-top"/>
                    <div className="card-body">
                    <h5 className="card-title">{movie.title}<span className="badge badge-secondary">{movie.release_date}</span></h5>
                    <h6 className="card-subtitle mb-2 text-muted">Card subtitle</h6>
                    </div>
                </StyledContainer>
            </Link>
        );
    }
}
