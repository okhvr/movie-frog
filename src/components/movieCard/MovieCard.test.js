import React from 'react';
import renderer from 'react-test-renderer';
import MovieCard from './MovieCard';
import { BrowserRouter } from "react-router-dom";

describe('MovieCard component', () => {
    const movie = {
        id: '1',
        poster_path: 'https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg',
        title: 'test',
        release_date: '12/06/1997'
    };
    it('should match with snapshot', () => {
        const component = renderer.create(
            <BrowserRouter>
                <MovieCard movie={movie}/>
            </BrowserRouter>
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});