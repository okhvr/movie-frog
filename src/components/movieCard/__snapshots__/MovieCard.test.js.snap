// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard component should match with snapshot 1`] = `
<a
  href="/movie/1"
  onClick={[Function]}
>
  <div
    className="card margins"
    style={
      Object {
        "width": "18rem",
      }
    }
  >
    <img
      className="card-img-top"
      src="https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg"
    />
    <div
      className="card-body"
    >
      <h5
        className="card-title"
      >
        test
        <span
          className="badge badge-secondary"
        >
          12/06/1997
        </span>
      </h5>
      <h6
        className="card-subtitle mb-2 text-muted"
      >
        Card subtitle
      </h6>
    </div>
  </div>
</a>
`;
